guava/src/com/google/common/reflect/ClassPath.java

```diff
@@ -478,10 +484,18 @@ protected void scanJarFile(ClassLoader classloader, JarFile file) {
 
     @Override
     protected void scanDirectory(ClassLoader classloader, File directory) throws IOException {
-      scanDirectory(directory, classloader, "");
+      scanDirectory(directory, classloader, "", new HashSet<>());
     }
 
-    private void scanDirectory(File directory, ClassLoader classloader, String packagePrefix)
+    /**
+     * Recursively scan the given directory, adding resources for each file encountered.  Symlinks which have already
+     * been traversed in the current tree path will be skipped to eliminate cycles; otherwise symlinks are traversed.
+     * @param directory the root of the directory to scan
```

Comment:
A blank line between the last paragraph and the first `@param` is required AFAIK. Use google-java-format to fix this.